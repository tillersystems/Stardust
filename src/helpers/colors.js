import { lighten, darken, setLightness } from 'polished';

export const dark = color => darken(0.03, color);

export const light = color => lighten(0.1, color);

export const whiteOpacity = opacity => `hsla(0, 100%, 100%, ${opacity})`;

export const getLigntness = color => {
  const colorWithoutPar = color.replace(/[hsl(%)]/g, '');
  const hslValue = colorWithoutPar.split(',');
  const lightessValue = hslValue[2];
  return lightessValue / 100;
};

export const setColorDark = (color, light) => {
  return setLightness(getLigntness(color) - light, color);
};
