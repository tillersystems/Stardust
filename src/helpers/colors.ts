import { lighten, darken, setLightness } from 'polished';

/**
 * Colors helper
 */

export const dark = (color: string) => darken(0.03, color);

export const light = (color: string) => lighten(0.1, color);

export const whiteOpacity = (opacity: number) => `hsla(0, 100%, 100%, ${opacity})`;

export const getLightness = (color: string) => {
  const colorWithoutPar = color.replace(/[hsl(%)]/g, '');
  const hslValue = colorWithoutPar.split(',');
  const lightessValue = Number(hslValue[2]);

  return lightessValue / 100;
};

export const setColorDark = (color: string, light: number) => {
  return setLightness(getLightness(color) - light, color);
};
