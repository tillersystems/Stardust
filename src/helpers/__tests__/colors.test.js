import 'jest-styled-components';

import { dark, light, whiteOpacity, getLightness, setColorDark } from '../colors';

describe('Colors helpers', () => {
  it('should render a darker color', () => {
    const expectedValue = dark('hsl(6, 79%, 65%)');
    expect(expectedValue).toMatchSnapshot();
  });
  it('should render a lighter color', () => {
    const expectedValue = light('hsl(6, 79%, 65%)');
    expect(expectedValue).toMatchSnapshot();
  });
  it('should render a white with an opacity', () => {
    const expectedValue = whiteOpacity(0.1);
    expect(expectedValue).toMatchSnapshot();
  });
  it('should render the lightness value of an hsl color', () => {
    const expectedValue = getLightness('hsl(6, 79%, 65%)');
    expect(expectedValue).toMatchSnapshot();
  });
  it('should render a darker light on hsl color', () => {
    const expectedValue = setColorDark('hsl(6, 79%, 65%)', 0.1);
    expect(expectedValue).toMatchSnapshot();
  });
});
