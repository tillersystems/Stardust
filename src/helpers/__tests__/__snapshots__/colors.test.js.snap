// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors helpers should render a darker color 1`] = `"#ea6151"`;

exports[`Colors helpers should render a darker light on hsl color 1`] = `"#e74432"`;

exports[`Colors helpers should render a lighter color 1`] = `"#f1978d"`;

exports[`Colors helpers should render a white with an opacity 1`] = `"hsla(0, 100%, 100%, 0.1)"`;

exports[`Colors helpers should render the lightness value of an hsl color 1`] = `0.65`;
